Studies/source/GenStudy.cxx

```diff
@@ -84,148 +97,124 @@ class GenStudy {
+        for(size_t plot_index = 1; plot_index < eff_plots.size(); ++plot_index){
             eff_plots.at(plot_index).Draw("SAME");
         }
     }
 
-    bool isCorrectChannel(const HHGenEvent& hh_event, Channel channel)
+    static Channel genChannel(const HHGenEvent& hh_event)
     {
-        if(channel == Channel::ETau){
-            if((hh_event.tau_decay.at(0) == GenDecayMode::Electron &&  hh_event.tau_decay.at(1)  == GenDecayMode::Hadrons) ||
-                    (hh_event.tau_decay.at(1) == GenDecayMode::Electron &&  hh_event.tau_decay.at(0)  == GenDecayMode::Hadrons))
-                return true;
-        }
-        else if(channel == Channel::MuTau){
-            if((hh_event.tau_decay.at(0) == GenDecayMode::Muon &&  hh_event.tau_decay.at(1)  == GenDecayMode::Hadrons) ||
-                    (hh_event.tau_decay.at(1) == GenDecayMode::Muon &&  hh_event.tau_decay.at(0)  == GenDecayMode::Hadrons))
-                return true;
-        }
-        else if(channel == Channel::TauTau){
-            if((hh_event.tau_decay.at(0) == GenDecayMode::Hadrons &&  hh_event.tau_decay.at(1)  == GenDecayMode::Hadrons))
-                return true;
-        }
-        return false;
+        if((hh_event.tau_decay.at(0) == TauGenDecayMode::Electron &&  hh_event.tau_decay.at(1)  == TauGenDecayMode::Hadrons) ||
+                (hh_event.tau_decay.at(1) == TauGenDecayMode::Electron &&  hh_event.tau_decay.at(0)  == TauGenDecayMode::Hadrons))
+            return Channel::ETau;
+
+        else if((hh_event.tau_decay.at(0) == TauGenDecayMode::Muon &&  hh_event.tau_decay.at(1)  == TauGenDecayMode::Hadrons) ||
+                    (hh_event.tau_decay.at(1) == TauGenDecayMode::Muon &&  hh_event.tau_decay.at(0)  == TauGenDecayMode::Hadrons))
+            return Channel::MuTau;
+
+        else if((hh_event.tau_decay.at(0) == TauGenDecayMode::Hadrons &&  hh_event.tau_decay.at(1)  == TauGenDecayMode::Hadrons))
+            return Channel::TauTau;
+//        else
+//            throw exception ("Unsupported channel.");
     }
 
-
-    bool isTauInsideAcceptance(const HHGenEvent& hh_event, Channel channel)
+    static bool isTauInsideAcceptance(const HHGenEvent& hh_event, Channel channel)
     {
-        //vector[0]=tau_lep_cut, vector[1]=tau_had_cut,
-        static const std::map<Channel, std::vector<int>> channel_pt_cuts = {
-            { Channel::ETau, {24,20} },
-            { Channel::MuTau, {20,20} },
-            { Channel::TauTau, {35,35} } };
+        static const std::map< Channel, std::vector<std::vector<int>> > channel_pt_cuts = {
+            { Channel::ETau, { {34,20}, {26,35} } },
+            { Channel::MuTau, { {25,20}, {21,32} } },
+            { Channel::TauTau,{ {40,40} } } };
+
+        for(size_t n = 0; n < channel_pt_cuts.at(channel).size(); ++n){
+            auto pt_cuts= channel_pt_cuts.at(channel).at(n);
+            if(hh_event.vis_tau[0].Pt() < pt_cuts.at(0) || hh_event.vis_tau[1].Pt() < pt_cuts.at(1))
+                return false;
```

Comment:
`false` should be returned only if all cut configurations are failed. So, the code should be:
```
bool pt_cuts_passed = false;
for(size_t n = 0; n < channel_pt_cuts.at(channel).size(); ++n){
    auto pt_cuts= channel_pt_cuts.at(channel).at(n);
    if(hh_event.vis_tau[0].Pt() > pt_cuts.at(0) && hh_event.vis_tau[1].Pt() > pt_cuts.at(1)) {
        pt_cuts_passed = true;
        break;
    }
}
if(!pt_cuts_passed)
    return false;
```